### PR TITLE
added packio

### PR DESCRIPTION
**Pull request by qchateau**
Specify library name and version:  **packio/1.0.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/packio/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        compiler, version = self.settings.compiler, Version(self.settings.compiler.version)
+        return any(compiler == sc[0] and version >= sc[1] for sc in supported_compilers)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "packio-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        if not self._supports_cpp17():
```
Comment (uilianries) [inline recipes/packio/all/conanfile.py]:
```suggestion
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, "17")
        elif not self._supports_cpp17():
```

Comment (uilianries) [reply recipes/packio/all/conanfile.py]: check cppstd when available, otherwise, check the compiler version.

Comment (Nipheris) [reply recipes/packio/all/conanfile.py]: @uilianries Seems that compiler check is [already done by `check_min_cppstd`](https://github.com/conan-io/conan/blob/410be07a8666270ea79c0db181a48127251d1673/conans/client/tools/settings.py#L49), isn't it? I think the whole method should be removed and `check_min_cppstd` is already enough (the exception is raised too).

Comment (qchateau) [reply recipes/packio/all/conanfile.py]: `check_min_cppstd` seem to fallback on the _default_ `cppstd` of the compiler, but that does not mean that the compiler does not support the required `cppstd`.

Or am I wrong ?

Comment (uilianries) [reply recipes/packio/all/conanfile.py]: > `check_min_cppstd` seem to fallback on the _default_ `cppstd` of the compiler, but that does not mean that the compiler does not support the required `cppstd`.
> 
> Or am I wrong ?

It supports partial or mostly complete. https://en.cppreference.com/w/cpp/compiler_support

Comment (uilianries) [reply recipes/packio/all/conanfile.py]: > @uilianries Seems that compiler check and supported cpp standard deduction is [already done by `check_min_cppstd`](https://github.com/conan-io/conan/blob/410be07a8666270ea79c0db181a48127251d1673/conans/client/tools/settings.py#L49), isn't it? I think the whole method should be removed and `check_min_cppstd` is already enough (the exception is raised too).

No. check_min_cppstd checks the default cppstd supported, not the required.

Comment (qchateau) [reply recipes/packio/all/conanfile.py]: So is the current check the right one or do I need to change something ?

Comment (uilianries) [reply recipes/packio/all/conanfile.py]: It's okay, but I would recommend you follow this approach: https://github.com/conan-io/conan-center-index/blob/fb0a2b23e8978397a436fef3066e11117f7af66d/recipes/entt/3.x.x/conanfile.py#L30

Better messages and warning.

Comment (Nipheris) [reply recipes/packio/all/conanfile.py]: > No. check_min_cppstd checks the default cppstd supported, not the required.

From the docs to `check_min_cppstd`:
> Check if current cppstd fits the minimal version required.

And `entt` library you mentioned uses exactly the same approach I'm talking about) - just use `check_min_cppstd` to require minimum C++ standard, and do not reinvent the wheel.

**Comment by qchateau** [comment]
Up, anything I can do ?

**Pull request merged**